R/getPrevalence.R

```diff
@@ -590,19 +594,46 @@ setMethod("agglomerateByPrevalence", signature = c(x = "SummarizedExperiment"),
             if(!is.null(rank)){
                 rowData(other_x)[,rank] <- other.label
             }
-            # temporary fix until TSE supports rbind
-            class <- c("SingleCellExperiment","RangedSummarizedExperiment",
-                       "SummarizedExperiment")
-            class_x <- class(x)
-            if(!(class_x %in% class)){
-                class <- "SingleCellExperiment"
-            } else {
-                class <- class[class_x == class]
-            }
-            x <- rbind(as(x[f,],class),
-                       as(other_x,class))
-            x <- as(x,class_x)
+            x <- rbind(x[f,], other_x)
         }
         x
     }
 )
+
+#' @rdname agglomerate-methods
+#' @export
+setMethod("agglomerateByPrevalence", 
+          signature = c(x = "TreeSummarizedExperiment"),
+    function(x, rank = NULL, other.label = other_label, other_label = "Other",
+            agglomerate.tree = FALSE, ...){
+      # input check
+      if(!.is_a_bool(agglomerate.tree)){
+        stop("'update.tree' must be TRUE or FALSE.", call. = FALSE)
+      }
+      # for optionally merging referenceSeq
+      f <- rownames(x) 
+      refSeq <- NULL
+      # update.refseq is a hidden parameter as for all other agglomeration methods
+      # from the agglomerate-methods man page.
+      # Here 'list(...)[["update.refseq"]]' is used to access it.
+      merge_refseq <- list(...)[["update.refseq"]]
+      if( is.null(merge_refseq) ){
+        merge_refseq <- FALSE
+      }
+      if( !.is_a_bool(merge_refseq) ){
+        stop("'update.refseq' must be TRUE or FALSE.", call. = FALSE)
+      }
+      if( merge_refseq ){
+        refSeq <- referenceSeq(x)
+      }
+      if( merge_refseq ){
+        referenceSeq(x) <- .merge_refseq_list(refSeq, f, rownames(x), ...)
+      }
+      x <- callNextMethod()
```

Comment:
You should probably first call

x <- callNextMethod()

and then

 if( merge_refseq ){
        referenceSeq(x) <- .merge_refseq_list(refSeq, f, rownames(x), ...)
 }
 
 I have not tested the code yet, but it looks like it